test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -0,0 +1,310 @@
+	// f must run inside an It or Context callback.
+	f := framework.NewFrameworkWithCustomTimeouts("volume-modify", storageframework.GetDriverTimeouts(driver))
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	init := func(ctx context.Context, createVolumeWithVAC bool) {
+		l = local{}
+
+		// TODO for reviewers - are the group/version/resource stored as a constant anywhere I can access? should I just make my own constants?
+		ok, err := discovery.IsResourceEnabled(f.ClientSet.Discovery(), schema.GroupVersionResource{Group: "storage.k8s.io", Version: "v1alpha1", Resource: "volumeattributesclasses"})
+		framework.ExpectNoError(err, "While checking for VolumeAttributesClass support")
+		if !ok {
+			e2eskipper.Skipf("API server does not support VolumeAttributesClasses - skipping")
+		}
+
+		l.config = driver.PrepareTest(ctx, f)
+		vacDriver, _ := driver.(storageframework.VolumeAttributesClassTestDriver)
+		l.vac = vacDriver.GetVolumeAttributesClass(ctx, l.config)
+
+		if l.vac == nil {
+			e2eskipper.Skipf("Driver %q returned nil VolumeAttributesClass - skipping", driver.GetDriverInfo().Name)
+		}
+
+		ginkgo.By("Creating VolumeAttributesClass")
+		_, err = f.ClientSet.StorageV1alpha1().VolumeAttributesClasses().Create(ctx, l.vac, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "While creating VolumeAttributesClass")
+
+		ginkgo.By("Creating volume")
+		testVolumeSizeRange := v.GetTestSuiteInfo().SupportedSizeRange
+		if createVolumeWithVAC {
+			l.resource = storageframework.CreateVolumeResourceWithVAC(ctx, driver, l.config, pattern, testVolumeSizeRange, &l.vac.Name)
+		} else {
+			l.resource = storageframework.CreateVolumeResource(ctx, driver, l.config, pattern, testVolumeSizeRange)
+		}
+	}
+
+	cleanup := func(ctx context.Context) {
+		var errs []error
+		if l.resource != nil {
+			ginkgo.By("Deleting VolumeResource")
+			errs = append(errs, l.resource.CleanupResource(ctx))
+			l.resource = nil
+		}
+
+		if l.vac != nil {
+			ginkgo.By("Deleting VAC")
+			errs = append(errs, CleanupVAC(ctx, l.vac, f.ClientSet, vacCleanupWaitPeriod))
+			l.vac = nil
+		}
+
+		framework.ExpectNoError(errors.NewAggregate(errs), "While cleaning up")
+	}
+
+	ginkgo.It("should create a volume with VAC", func(ctx context.Context) {
+		init(ctx, true /* volume created with VAC */)
+		ginkgo.DeferCleanup(cleanup)
+
+		ginkgo.By("Creating a pod with dynamically provisioned volume")
+		podConfig := e2epod.Config{
+			NS:            f.Namespace.Name,
+			PVCs:          []*v1.PersistentVolumeClaim{l.resource.Pvc},
+			SeLinuxLabel:  e2epod.GetLinuxLabel(),
+			NodeSelection: l.config.ClientNodeSelection,
+			ImageID:       e2epod.GetDefaultTestImageID(),
+		}
+		pod, err := e2epod.CreateSecPodWithNodeSelection(ctx, f.ClientSet, &podConfig, f.Timeouts.PodStart)
+		ginkgo.DeferCleanup(e2epod.DeletePodWithWait, f.ClientSet, pod)
+		framework.ExpectNoError(err, "While creating test pod with VAC")
```

Comment:
Added an extra check at the end of this test that `vacMatches(pv, pvc)` (which checks all those values) is `true`

Reply:
Added a check for `pvc.status.currentVAC` and `pv.spec.vacName`, but had to workaround `pvc.status.volumeModifyStatus` because the current `external-provisioner` implementation does not set it for volumes freshly provisioned with a VAC. Not sure if that should be considered a bug in the provisioner?

Reply:
The external-resizer is responsible for updating this field when it observes the pvc and pv's changed. The external-provsioner doesn't call the `ControllerModifyVolume ` PRC. when a pvc is created with a given vac. it will call the `CreateVolume` PRC that accepts mutable parameters in creation process. And external-provisioner is responsible for provisioning a volume. It doesn't modify pvc's status. when a pv is created, the pv will be observed by the kube-controller-manager and external-resizer. 

The kube-controller-manager won't be able to change the currentVACName of a pvc the util https://github.com/kubernetes/kubernetes/pull/121902 is merged. so the external-resizer is only one. I'm not sure whether the external-resizer should do that. If it also does, there's a race between kube-controller-manager and external-resizer.

During the pvc is created with a vac, the external-resizer shouldn't call `ControllerModifyVolume` and volumeModifyStatus should be nil because the vac is assigned to pv. Do we need to add an e2e test for it?

/cc @msau42 @xing-yang @jsafrane

Reply:
volumeModifyStatus is only set if a change is made. During provisioning, it should be created with the correct VAC so the `volumeModifyStatus` should be unset.

Reply:
Because this is intended, I added a parameter to `vacMatches` and added a comment explaining we only should check current status when the volume is modified because we don't expect it to be set when creating.